test/converters/xcresult/xcresult.go

```diff
@@ -0,0 +1,188 @@
+	FormatVersion     string
+	TestableSummaries []TestableSummary
+}
+
+func collapseSubtestTree(data Subtests) (tests Subtests) {
+	for _, test := range data {
+		if len(test.Subtests) > 0 {
+			tests = append(tests, collapseSubtestTree(test.Subtests)...)
+		}
+		if test.TestStatus != "" {
+			tests = append(tests, test)
+		}
+	}
+	return
+}
+
+func (summaryPlist TestSummaryPlist) Tests() map[string]Subtests {
+	tests := map[string]Subtests{}
+	var subTests Subtests
+	for _, testableSummary := range summaryPlist.TestableSummaries {
+		for _, test := range testableSummary.Tests {
+			subTests = append(subTests, collapseSubtestTree(test.Subtests)...)
+		}
+	}
+	for _, test := range subTests {
+		testID := strings.Split(test.TestIdentifier, "/")[0]
+		tests[testID] = append(tests[testID], test)
+	}
+	return tests
+}
+
+type TestableSummary struct {
+	TargetName      string
+	TestKind        string
+	TestName        string
+	TestObjectClass string
+	Tests           []Test
+}
+
+type FailureSummary struct {
+	FileName           string
+	LineNumber         int
+	Message            string
+	PerformanceFailure bool
+}
+
+type Subtest struct {
+	Duration         float64
+	TestStatus       string
+	TestIdentifier   string
+	TestName         string
+	TestObjectClass  string
+	Subtests         Subtests
+	FailureSummaries []FailureSummary
+}
+
+func (st Subtest) Failure() (message string) {
+	prefix := ""
+	for _, failure := range st.FailureSummaries {
+		message += fmt.Sprintf("%s%s:%d - %s", prefix, failure.FileName, failure.LineNumber, failure.Message)
+		prefix = "\n"
+	}
+	return
+}
+
+type Subtests []Subtest
+
+func (sts Subtests) FailuresCount() (count int) {
+	for _, test := range sts {
+		if len(test.FailureSummaries) > 0 {
+			count++
+		}
+	}
+	return count
+}
+
+func (sts Subtests) TotalTime() (time float64) {
+	for _, test := range sts {
+		time += test.Duration
+	}
+	return time
+}
+
+type Test struct {
+	Subtests Subtests
+}
+
+//
+////
+
+// Junit4 XML
+
+type Junit4XML struct {
+	XMLName    xml.Name `xml:"testsuites"`
+	TestSuites []TestSuite
+}
+
+type TestSuite struct {
+	XMLName   xml.Name `xml:"testsuite"`
+	Name      string   `xml:"name,attr"`
+	Tests     int      `xml:"tests,attr"`
+	Failures  int      `xml:"failures,attr"`
+	Errors    int      `xml:"errors,attr"`
+	Time      float64  `xml:"time,attr"`
+	TestCases []TestCase
+}
+
+type TestCase struct {
+	XMLName   xml.Name `xml:"testcase"`
+	Name      string   `xml:"name,attr"`
+	ClassName string   `xml:"classname,attr"`
+	Time      float64  `xml:"time,attr"`
+	Failure   string   `xml:"failure,omitempty"`
+}
+
+//
+////
+
+type Handler struct {
+	files                  []string
+	testSummariesPlistPath string
+}
+
+func (h *Handler) SetFiles(files []string) {
+	h.files = files
+}
+func (h *Handler) Detect() bool {
+	for _, file := range h.files {
+		if strings.HasSuffix(file, ".xcresult") {
+			h.testSummariesPlistPath = filepath.Join(file, "TestSummaries.plist")
```

Comment:
Should be fine, good to know however that subfolders containing different build phases contain TestSummaries.plist version 1.2 with additional info about the running environment (os version, device etc.), instead of 1.1 in the root.

Reply:
Could you please explain this a bit more?

Reply:
See [](https://michele.io/test-logs-in-xcode/). There are more than 1 plists, mostly similar content.

Reply:
I see, however I think it is the main goal to list all test cases nevertheless in whichbuild phases they are.